# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/firefox/hotfix-gcc14.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- firefox-136.0/media/ffvpx/libavcodec/av1dec.c.vanilla	2025-03-04 01:32:46.194009026 +0100
+++ firefox-136.0/media/ffvpx/libavcodec/av1dec.c	2025-03-04 01:33:05.323855594 +0100
@@ -904,7 +904,7 @@
         ff_cbs_fragment_reset(&s->current_obu);
     }
 
-    s->dovi.logctx = avctx;
+    s->dovi.logctx = (AVContext *)avctx;
     s->dovi.cfg.dv_profile = 10; // default for AV1
     sd = ff_get_coded_side_data(avctx, AV_PKT_DATA_DOVI_CONF);
     if (sd && sd->size >= sizeof(s->dovi.cfg))
--- firefox-136.0/media/ffvpx/libavcodec/libdav1d.c.vanilla	2025-03-04 01:33:28.230510736 +0100
+++ firefox-136.0/media/ffvpx/libavcodec/libdav1d.c	2025-03-04 01:33:39.159180563 +0100
@@ -293,7 +293,7 @@
     c->delay = res > 1 ? res : 0;
 #endif
 
-    dav1d->dovi.logctx = c;
+    dav1d->dovi.logctx = (AVContext *)c;
     dav1d->dovi.cfg.dv_profile = 10; // default for AV1
     sd = ff_get_coded_side_data(c, AV_PKT_DATA_DOVI_CONF);
     if (sd && sd->size >= sizeof(dav1d->dovi.cfg))
